### DISCUSSION TITLE
What is the rationale for ErrorOnEmailAsUsernameWithDuplicateEmailCase

### DISCUSSION THREAD

**Discussion by aaww01** [authelia/authelia#5654] [General]
I need some help understanding the rationale for the following restriction...

When authelia's file authentication provider is configured to run with 
file.search.email: false
file.search.case_insensitive: true
There is a validation that is performed on startup, that if it fails causes Authelia to not start.
In the file_user_provider_test.go, there is this explanation:
**error loading authentication database: username 'john.doe@authelia.com' is configured as an email for user with username 'john' which isn't allowed when case-insensitive search is enabled**

Based on the following test case, what is the rationale behind not allowing a username and a different user's email address to overlap given that the system is set to **not** search email addresses, thus a user logging on as 'john.doe@authelia.com' is not the same user logging on as 'john'. 

```users:
  john.doe@authelia.com:
    displayname: "John Doe"
    password: "{CRYPT}$argon2id$v=19$m=65536,t=3,p=2$BpLnfgDsc2WD8F2q$o/vzA4myCqZZ36bUGsDY//8mKUYNZZaR0t4MFFSs+iM"
    email: JOHN@authelia.com
    groups:
      - admins
      - dev

  john:
    displayname: "John Potter"
    password: "{CRYPT}$6$rounds=500000$jgiCMRyGXzoqpxS3$w2pJeZnnH8bwW3zzvoMWtTRfQYsHbWbD/hquuQ5vUeIyl9gdwBIt6RWk2S6afBA0DPakbeWgD/4SZPiS0hYtU/"
    email: john.doe@authelia.com
    groups: []```

**Comment by james-d-elliott** [comment]
I'm pretty sure the rationale at the time is it's much simpler logic to implement. What's the rationale behind having this confusing mix of users?
